ElectronClient/gui/NoteEditor/NoteBody/CodeMirror/Editor.tsx

```diff
@@ -20,13 +20,16 @@ import 'codemirror/keymap/sublime'; // Used for swapLineUp and swapLineDown
 
 import 'codemirror/mode/gfm/gfm';
 import 'codemirror/mode/xml/xml';
+import 'codemirror/mode/meta';
 // Modes for syntax highlighting inside of code blocks
-import 'codemirror/mode/python/python';
-import 'codemirror/mode/javascript/javascript';
-import 'codemirror/mode/markdown/markdown';
-import 'codemirror/mode/clike/clike';
-import 'codemirror/mode/diff/diff';
-import 'codemirror/mode/sql/sql';
+// Load all modes
+for (let i = 0; i < CodeMirror.modeInfo.length; i++) {
+	const mode = CodeMirror.modeInfo[i].mode;
+	// plaintext has null mode for some reason
+	if (mode !== 'null') {
+		require(`codemirror/mode/${mode}/${mode}`);
```

Comment:
Using a very unscientific method (console.log before and after the loop and comparing timestamps) this takes about 100ms to load all modes.

It's an interesting question. The only code that gets run on require should be `CodeMirror.defineMode` and maybe `CodeMirror.defineMime`. Of course it's always possible that some side effect will be introduced by an update, I'm willing to trust the Code mirror merge and testing process that this won't actually happen in practice. 

An alternative would be to add some code that parses code blocks and loads the necessary mode at runtime, but in practice I don't believe this will be any safer and it trades off startup performance for runtime.

One of the benefits of running this on load is that if a bug is introduced in a codemirror update (I don't believe this will happen), then we'll catch it as soon as we update because Joplin will either crash or log an error which will allow us to rollback the upgrade until it's fixed.

Reply:
100ms feels like quite a lot actually, how many modes are we loading here? What I'm thinking is that if we support the top languages, it might load in 5ms and cover 95% of use cases. So by loading everything we spend the largest chunk of the time loading modules that almost no-one will need.